src/stream/Middleware.ts

```diff
@@ -169,6 +180,11 @@ export class OrderedHistoryResumer extends MiddlewareBase {
     // This is sufficiently low likelihood that this can be ignored for now.
     this.historicalMessages.sort((a, b) => this.reverseOrderer(a.id, b.id));
 
+    // A sequenceId of 0 means there is no historical message to seek to. All history should be applied after paging complete
```

Comment:
The logic is somewhat complex, so worth adding a clarifying comment.

When we have rewind-by-application-id this will be much simpler.

```suggestion
    // A sequenceId of 0 is a cursor that represents the position before the start of the stream.
    // There is no such historical message to seek to, instead we should paginate through all
    // of history to obtain all messages that were published since the sequenceId was obtained.
```